src/users/index.js

```diff
@@ -61,6 +61,14 @@ class UserProvider {
    * @returns {PromiseLike<U | undefined> | U | undefined }
    */
   getUser (userId) { throw new Error('unimplemented') }
+
+  /**
+   * Adds a new user to this user provider
+   *
+   * @param {string} userId The id of the user to add
+   * @returns {PromiseLike<U | undefined> | U | undefined }
```

Comment:
This should probably be `@returns {PromiseLike<U> | U}` since I wouldn't really expect this to fail to produce a user

Reply:
This also allows errors here https://github.com/codeforlansing/cityzen-server/blob/e48bfbc617abaff4768faa8a00fbdd301dac79db/src/users/index.js#L95-L99 

without a second null check

Reply:
Agree! Let's clean this up when we do a DB implementation. 👍